cardano-c/src/key.rs

```diff
@@ -15,22 +15,47 @@ pub extern "C" fn cardano_xprv_derive(c_xprv: XPrvPtr, index: u32) -> XPrvPtr {
 }
 
 #[no_mangle]
-pub extern "C" fn cardano_xprv_from_bytes(c_xprv: *const u8) -> XPrvPtr {
+pub extern "C" fn cardano_xprv_from_bytes(
+    c_xprv: *const u8,
+    xprv_out: *mut XPrvPtr,
+) -> CardanoResult {
     let xprv_data = unsafe { slice::from_raw_parts(c_xprv, hdwallet::XPRV_SIZE) };
-    match hdwallet::XPrv::from_slice(xprv_data) {
+    let array = {
+        let mut array = [0u8; 96];
+        array.copy_from_slice(xprv_data);
+        array
+    };
+    match hdwallet::XPrv::from_bytes_verified(array) {
         Ok(r) => {
+            println!("ok");
             let xprv = Box::new(r);
-            Box::into_raw(xprv)
-        },
-        Err(_) => {
-            ptr::null_mut()
+            unsafe { ptr::write(xprv_out, Box::into_raw(xprv)) };
+            CardanoResult::success()
         }
+        Err(_) => CardanoResult::failure(),
     }
 }
 
 #[no_mangle]
 pub extern "C" fn cardano_xprv_to_bytes(c_xprv: XPrvPtr) -> *const u8 {
-    unimplemented!()
+    //Get the inner byte array without taking ownership
+    let slice: &[u8] = unsafe { (*c_xprv).as_ref() };
+
+    let mut vector: Vec<u8> = Vec::with_capacity(hdwallet::XPRV_SIZE);
+    vector.extend_from_slice(slice);
+
+    //Get pointer to the inner value
+    let ptr = vector.as_ptr();
+
+    //Avoid running the destructor
+    std::mem::forget(vector);
+    ptr
+}
+
+#[no_mangle]
+pub extern "C" fn cardano_xprv_bytes_delete(bytes: *mut u8) {
+    let vector = unsafe { Vec::from_raw_parts(bytes, hdwallet::XPRV_SIZE, hdwallet::XPRV_SIZE) };
+    std::mem::drop(vector)
```

Comment:
Would be nice to zero the value here. We have a function for this in [`cardano::util::securemem::zero`](https://github.com/input-output-hk/rust-cardano/blob/master/cardano/src/util/securemem.rs#L4) that will do the work.

Simply add:

```rust
cardano::util::securemem::zero(&mut vector);
```

Reply:
Done!